metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/AbstractMetisPluginMetadata.java

```diff
@@ -15,8 +15,8 @@
 @JsonSubTypes({
     @JsonSubTypes.Type(value=OaipmhHarvestPluginMetadata.class, name="OAIPMH_HARVEST"),
     @JsonSubTypes.Type(value=HTTPHarvestPluginMetadata.class, name="HTTP_HARVEST"),
-    @JsonSubTypes.Type(value=DereferencePluginMetadata.class, name="DEREFERENCE"),
-    @JsonSubTypes.Type(value=VoidMetisPluginMetadata.class, name="VOID")
+    @JsonSubTypes.Type(value=EnrichmentPluginMetadata.class, name="ENRICHMENT"),
+    @JsonSubTypes.Type(value=ValidationPluginMetadata.class, name="VALIDATION")
 })
 public interface AbstractMetisPluginMetadata {
```

Comment:
Same comment as for class AbstractMetisPlugin: Implementing classes are very similar! Maybe we can introduce an abstract class that implements most of the functionality? Also this interface should be renamed, but I am not sure if this is feasible (it is implemented in the metis-core-workflow-qa dependency).

Reply:
Yes indeed I have in mind, just waiting until we have some more plugins ready to see how that looks and probably an abstract class is the way to go.